src/Bicep.Core/TypeSystem/TypeValidator.cs

```diff
@@ -38,12 +38,14 @@ public static IList<Diagnostic> GetCompileTimeConstantViolation(SyntaxBase expre
         /// <returns>Returns true if values of the specified source type are assignable to the target type. Returns false otherwise or null if assignability cannot be determined.</returns>
         public static bool? AreTypesAssignable(TypeSymbol? sourceType, TypeSymbol? targetType)
         {
-            if (sourceType == null || targetType == null || sourceType is ErrorTypeSymbol || targetType is ErrorTypeSymbol)
+            // return null when sourceType, targetType is either null or sourceType is UnassignableTypeSymbol and contains errors -> GetDiagnostics() returns a list of errors or when targetType is UnassignableTypeSymbol
+            if (sourceType == null || targetType == null || ( sourceType is UnassignableTypeSymbol unassignableTypeSymbol && unassignableTypeSymbol.GetDiagnostics().Any() ) || targetType is UnassignableTypeSymbol)
```

Comment:
Shouldn't we always return `null` if the source type is `UnassignableTypeSymbol`? What situations are there where target type is `UnassignableTypeSymbol`?

Reply:
TargetType I don't know, but left this as it was before.

@majastrz thoughts?

Reply:
The error type is not assignable to a different or even same error type regardless of how many diagnostics it contains, so leaving it like it was is the right thing to do.